sdks/java/core/src/main/java/org/apache/beam/sdk/testing/CoderProperties.java

```diff
@@ -322,6 +323,26 @@
     }
   }
 
+  public static <T> void coderConsumesBytesEqualToBytesEncoded(Coder<T> coder, T value)
```

Comment:
We could update all our existing users automatically if we fold this logic into the decode(Coder<T>, Context, bytes[]) method.

There we can check to see if the context is nested and then pad the byte array before decoding.